### PR TITLE
return notfound errors w/o wrappering them

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
@smarterclayton ptal

File: pkg/build/generator/generator.go [hunk 1]
```diff
@@ -233,6 +233,9 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 
 	newBuild, err := g.generateBuildFromConfig(ctx, bc, request.Revision, request.Binary)
 	if err != nil {
+		if errors.IsNotFound(err) {
```
Comment (smarterclayton) [inline pkg/build/generator/generator.go]: All of the other API types should be ignored.  generateBuildFromConfig is responsible for generating the correct API error type - if you want to transform it, transform it in generateBuildFromConfig.  We should not be wrapping any error that comes out of gbfc unless it's not an API error.

Comment (bparees) [inline pkg/build/generator/generator.go]: updated.

File: pkg/build/generator/generator.go [hunk 2]
```diff
@@ -233,6 +234,9 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 
 	newBuild, err := g.generateBuildFromConfig(ctx, bc, request.Revision, request.Binary)
 	if err != nil {
+		if oerrors.IsKnownAPIErrorType(err) {
```
Comment (smarterclayton) [inline pkg/build/generator/generator.go]: The correct test is casting to APIStatus in the kube `pkg/api/errors` package - if the error implements that, we consider it already handled.

**Comment by bparees** [comment]
@smarterclayton updated.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**